backend/data_tools/src/disable_users/disable_users.py

```diff
@@ -79,12 +79,16 @@ def update_disabled_users_status(conn: sqlalchemy.engine.Engine):
         logger.info("Fetching inactive users.")
         results = []
         all_users = se.execute(select(User)).scalars().all()
+        cutoff_date = datetime.now() - timedelta(days=60)
         for user in all_users:
             latest_session = get_latest_user_session(user_id=user.id, session=se)
-            if user.status == UserStatus.ACTIVE and latest_session and latest_session.last_active_at < datetime.now() - timedelta(days=60):
-                results.append(user.id)
-            elif latest_session is None and user.status == UserStatus.ACTIVE and (user.updated_on < datetime.now() - timedelta(days=60)):
-                results.append(user.id)
+            if user.status == UserStatus.ACTIVE:
+                stale_user = user.updated_on < cutoff_date
+                stale_session = latest_session and latest_session.last_active_at < cutoff_date
+                never_logged_in = latest_session is None
+
+                if (never_logged_in and stale_user) or (stale_session or stale_user):
```

Comment:
Shouldn't the logic be:
`if (never_logged_in and stale_user) or (stale_session and stale_user):`

That would test that the user last had an active session more than 60 days ago and the user has been updated or activated in the last 60 days.

Reply:
@johndeange I think you're right.  updated.